### PR TITLE
Docs async

### PR DESCRIPTION
**Pull request by Hugheth**

### CODE REVIEW

File: spec/Async.spec.lua [hunk 1]
```diff
@@ -122,31 +197,142 @@ describe(
...
+				it(
+					"can timeout after a delay",
+					function()
+						advanceAndAssertPromiseRejects(Async.timeout(Async.delay(2), 1, "Expected Error"), "Expected Error")
+					end
+				)
+				it(
+					"can resolve within delay",
+					function()
+						advanceAndAssertPromiseResolves(Async.timeout(Async.delay(1):andThen(Functions.returns("Ok")), 2))
```
Comment (ckdavid) [inline spec/Async.spec.lua]: What's the `2`? It doesn't look like it is an "assertion" argument.

File: src/Async.lua [hunk 2]
```diff
@@ -30,10 +49,184 @@ function Async.all(objects)
...
+				array,
+				function(promise)
+					Async.finally(
+						promise,
+						function(ok, result)
+							if #results < n then
+								if ok then
+									table.insert(results, result)
+									if #results == n then
+										resolve(results)
```
Comment (ckdavid) [inline src/Async.lua]: For readability's sake, I'd reduce the level of nesting here.

File: src/Async.lua [hunk 3]
```diff
@@ -30,10 +49,184 @@ function Async.all(objects)
...
+		end
+		local toastie = _.props({
+			bread = "brown",
+			filling = heat("cheese")
+		})
+		toastie:await() --> {bread = "brown", filling = "hot-cheese"} (1 second later)
+	@usage Values which are not promises are considered resolved immediately.
+]]
+--: <T>((Promise<T> | T){}) -> Promise<T{}>
+function Async.props(dictionary)
```
Comment (ckdavid) [inline src/Async.lua]: `_.props` seems a fairly obscure name (from Bluebird). It doesn't suggest anything related to promises. Something like `parallelDict` would at least make the similarity with `parallel` clearer.

File: spec/Async.spec.lua [hunk 4]
```diff
@@ -92,15 +94,88 @@ describe(
...
+				)
+			end
+		)
+		describe(
+			"resolve",
+			function()
+				it(
+					"resolves for multiple return values",
+					function()
+						local andThen = spy.new()
```
Comment (ckdavid) [inline spec/Async.spec.lua]: Did you mean `stub.new()`?

File: tools/rodocs/generateMd.ts [hunk 5]
```diff
@@ -128,8 +131,34 @@ function ${libName}.${name}(${params.join(', ')}) --> string
...
+			lines.push('\n**Usage**\n');
+			lines.push(...formatList(throws));
+		}
+
+		const rejects = filterEntries(doc.entries, 'rejects');
+		if (rejects.length) {
+			lines.push('\n**Rejects**\n');
+			lines.push(
+				...formatList(rejects, function(line) {
+					switch (line) {
```
[the remark this answers is not in the document]
Comment (Hugheth) [reply tools/rodocs/generateMd.ts]: to be extensible to any other types of rejections added

File: src/Async.lua [hunk 6]
```diff
@@ -42,46 +235,103 @@ function Async.delay(delayInSeconds)
...
+	@example
+		local fetch = _.async(function( url )
+			local HttpService = game:GetService("HttpService")
+			return HttpService:GetAsync(url)
+		end)
+		_.props({
+			main = fetch("http://example.com/burger"),
+			side = fetch("http://example.com/fries") 
+		}):andThen(function( meal )
+			print("Meal", _.pretty(meal))
```
[the remark this answers is not in the document]
Comment (Hugheth) [reply src/Async.lua]: I'm adding this to a future PR

File: src/Async.lua (hunk 3 above)
Comment (Hugheth) [reply src/Async.lua]: I use `All` for dictionary args elsewhere so will keep this convention

**Pull request merged**